`tests/rustdoc-ui` fails on main with `profile = "tools"` due to normalization.

### Command used

```sh
./x test ./tests/rustdoc-ui/track-diagnostics.rs
```

### Expected behaviour

Test passes

### Actual behaviour

```
gh-aDotInTheVoid@dev-desktop-eu-2:~/rust0$ ./x test ./tests/rustdoc-ui/track-diagnostics.rs
Building bootstrap
    Finished `dev` profile [unoptimized] target(s) in 0.05s
/home/gh-aDotInTheVoid/rust0/build/x86_64-unknown-linux-gnu/ci-llvm/bin/llvm-strip does not exist; skipping copy
Creating a sysroot for stage2 compiler (use `rustup toolchain link 'name' build/host/stage2`)
Building stage1 compiletest (stage0 -> stage1, x86_64-unknown-linux-gnu)
    Finished `release` profile [optimized] target(s) in 0.13s
Testing stage2 with compiletest suite=rustdoc-ui mode=ui (x86_64-unknown-linux-gnu)

running 1 tests

[ui] tests/rustdoc-ui/track-diagnostics.rs ... F


failures:

---- [ui] tests/rustdoc-ui/track-diagnostics.rs stdout ----
Saved the actual stderr to `/home/gh-aDotInTheVoid/rust0/build/x86_64-unknown-linux-gnu/test/rustdoc-ui/track-diagnostics/track-diagnostics.stderr`
diff of stderr:

6          |              |
7          |              expected because of the type of the constant
8          |
-          = note: -Ztrack-diagnostics: created at compiler/rustc_trait_selection/src/error_reporting/infer/mod.rs:LL:CC
+          = note: -Ztrack-diagnostics: created at /rustc-dev/212ef7770dfad656782207fda799bdae28fc5b7b/compiler/rustc_trait_selection/src/error_reporting/infer/mod.rs:LL:CC
10
11      error: aborting due to 1 previous error
12

Note: some mismatched output was normalized before being compared
-          = note: -Ztrack-diagnostics: created at /rustc-dev/212ef7770dfad656782207fda799bdae28fc5b7b/compiler/rustc_trait_selection/src/error_reporting/infer/mod.rs:1993:35
+          = note: -Ztrack-diagnostics: created at /rustc-dev/212ef7770dfad656782207fda799bdae28fc5b7b/compiler/rustc_trait_selection/src/error_reporting/infer/mod.rs:LL:CC


The actual stderr differed from the expected stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args track-diagnostics.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/home/gh-aDotInTheVoid/rust0/build/x86_64-unknown-linux-gnu/stage2/bin/rustdoc" "/home/gh-aDotInTheVoid/rust0/tests/rustdoc-ui/track-diagnostics.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/home/gh-aDotInTheVoid/.cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/home/gh-aDotInTheVoid/rust0/vendor" "--sysroot" "/home/gh-aDotInTheVoid/rust0/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-o" "/home/gh-aDotInTheVoid/rust0/build/x86_64-unknown-linux-gnu/test/rustdoc-ui/track-diagnostics" "-A" "internal_features" "-A" "incomplete_features" "-A" "unused_parens" "-A" "unused_braces" "-Cdebuginfo=0" "-Z" "track-diagnostics"
stdout: none
--- stderr -------------------------------
error[E0308]: mismatched types
  --> /home/gh-aDotInTheVoid/rust0/tests/rustdoc-ui/track-diagnostics.rs:10:18
   |
LL | pub const S: A = B;
   |              -   ^ expected `A`, found `B`
   |              |
   |              expected because of the type of the constant
   |
   = note: -Ztrack-diagnostics: created at /rustc-dev/212ef7770dfad656782207fda799bdae28fc5b7b/compiler/rustc_trait_selection/src/error_reporting/infer/mod.rs:1993:35

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0308`.
------------------------------------------

---- [ui] tests/rustdoc-ui/track-diagnostics.rs stdout end ----

failures:
    [ui] tests/rustdoc-ui/track-diagnostics.rs

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 364 filtered out; finished in 262.86ms

Some tests failed in compiletest suite=rustdoc-ui mode=ui host=x86_64-unknown-linux-gnu target=x86_64-unknown-linux-gnu
Build completed unsuccessfully in 0:00:08
```

### Bootstrap configuration (bootstrap.toml)

```toml
# See bootstrap.example.toml for documentation of available options
#
profile = "tools"  # Includes one of the default files in src/bootstrap/defaults
change-id = 153143
```

### Operating system

Ubuntu 24.04.3 LTS

### HEAD

212ef7770dfad656782207fda799bdae28fc5b7b


### Additional context

The test already normalizes the output, we probably need to normalize it more. 

https://github.com/rust-lang/rust/blob/80d0e4be6f15899649ba31669077c59a986f96cc/tests/rustdoc-ui/track-diagnostics.rs#L2-L6

--- a/tests/rustdoc-ui/track-diagnostics.rs
+++ b/tests/rustdoc-ui/track-diagnostics.rs
@@ -3,6 +3,7 @@
 // Normalize the emitted location so this doesn't need
 // updating everytime someone adds or removes a line.
 //@ normalize-stderr: ".rs:\d+:\d+" -> ".rs:LL:CC"
+//@ normalize-stderr: "/rustc(?:-dev)?/[a-z0-9.]+/" -> ""
 
 struct A;
 struct B;

